src/util/memory_manager.h

```diff
@@ -28,7 +28,7 @@ Revision History:
 #endif
 
 #ifdef __GNUC__
-# if (__GNUC__ * 100 + __GNUC_MINOR__) >= 409 || __has_builtin(returns_nonnull)
+# if ((__GNUC__ * 100 + __GNUC_MINOR__) >= 409 || __has_builtin(returns_nonnull)) && !defined(__INTEL_COMPILER)
```

Comment:
Wouldn't this be more clear by having a separate block for `__INTEL_COMPILER`?

```c
#ifdef __INTEL_COMPILER
# define ALLOC_ATTR __attribute__((malloc))
#elif defined(__GNUC__)
  ...
#elif defined(_WINDOWS)
# define ALLOC_ATTR __declspec(restrict)
#else
#  define ALLOC_ATTR
#endif
```

Reply:
agree

Reply:
I specifically included this here, because `__INTEL_COMPILER` is only available when ICC is used on Linux or in GCC-emulation mode, i.e. when `__GNUC__` is defined. ICC impersonates different compilers based on the platform and exposes different macros in each of modes. I don't know which macro would be the best to use to cover all the cases and how far the impersonation goes, it might be possible this GCC-ism is not supported in MSVC emulation mode (and I have no way of finding out).